daemon/algod/api/server/v2/handlers.go

```diff
@@ -0,0 +1,420 @@
+	if err != nil {
+		return returnError(ctx, http.StatusBadRequest, err, err.Error(), v2.Log)
+	}
+
+	// For backwards compatibility, return txid of first tx in group
+	txid := txgroup[0].ID()
+	return ctx.JSON(http.StatusOK, generated.PostTransactionsResponse{TxId: txid.String()})
+}
+
+// TransactionParams gets parameters for constructing a new transaction
+// (GET /v2/transactions/params)
+func (v2 *Handlers) TransactionParams(ctx echo.Context) error {
+	stat, err := v2.Node.Status()
+	if err != nil {
+		return returnError(ctx, http.StatusInternalServerError, err, errFailedRetrievingNodeStatus, v2.Log)
+	}
+
+	gh := v2.Node.GenesisHash()
+
+	var params generated.TransactionParams
+	params.Fee = v2.Node.SuggestedFee().Raw
+	params.GenesisId = v2.Node.GenesisID()
+	params.GenesisHash = gh[:]
+	params.LastRound = uint64(stat.LastRound)
+	params.ConsensusVersion = string(stat.LastVersion)
+
+	proto := config.Consensus[stat.LastVersion]
+	params.MinFee = proto.MinTxnFee
+
+	return ctx.JSON(http.StatusOK, params)
+}
+
+// PendingTransactionInformation gets a specific pending transaction, or looks it up in the ledger if the recently
+// confirmed block is still in the ledger.
+// (GET /v2/transactions/pending/{txid})
+func (v2 *Handlers) PendingTransactionInformation(ctx echo.Context, txid string, params generated.PendingTransactionInformationParams) error {
+	txID := transactions.Txid{}
+	if err := txID.UnmarshalText([]byte(txid)); err != nil {
+		return returnError(ctx, http.StatusBadRequest, err, errNoTxnSpecified, v2.Log)
+	}
+
+	if txn, ok := v2.Node.GetPendingTransaction(txID); ok {
+		response := generated.PendingTransactionResponse{
+			Txn:             "",
+			PoolError:       "",
+			ClosingAmount:   nil,
+			ConfirmedRound:  nil,
+			SenderRewards:   nil,
+			ReceiverRewards: nil,
+			CloseRewards:    nil,
+		}
+
+		handle, err := getCodecHandle(params.Format)
+		if err != nil {
+			return returnError(ctx, http.StatusBadRequest, err, errFailedParsingFormatOption, v2.Log)
+		}
+
+		encoded, err := encode(handle, txn.Txn)
+		if err != nil {
+			return returnError(ctx, http.StatusInternalServerError, err, errFailedToParseTransaction, v2.Log)
+		}
+
+		response.Txn = encoded
+
+		if txn.ConfirmedRound != 0 {
+			r := uint64(txn.ConfirmedRound)
+			response.ConfirmedRound = &r
+
+			response.ClosingAmount = &txn.ApplyData.ClosingAmount.Raw
+			response.SenderRewards = &txn.ApplyData.SenderRewards.Raw
+			response.ReceiverRewards = &txn.ApplyData.ReceiverRewards.Raw
+			response.CloseRewards = &txn.ApplyData.CloseRewards.Raw
+
+			response.AssetIndex = computeAssetIndexFromTxn(txn, v2.Node.Ledger())
+		}
+
+		return ctx.JSON(http.StatusOK, response)
+	}
+
+	// We didn't find it, return a failure
+	err := errors.New(errTransactionNotFound)
+	return returnError(ctx, http.StatusNotFound, err, err.Error(), v2.Log)
+}
+
+// getPendingTransactions is a generalized version of the get pending transactions endpoints for code reuse.
```

Comment:
seems like this function should not be a function of `v2 *Handlers` (since it's not a handler). Consider making it an independent one and handle the responses in the callers.

Reply:
Yes, thats why I made it private.

Reply:
but it's not. It's still part of the Handlers struct.

Reply:
but it's not. It's still part of `Handlers`.

Reply:
I guess I'm not sure what your concern is here. It isn't visible outside of the `v2` package, if there was a way to reduce the scope further I would have done that.

In addition to not being visible outside of the package, it's also not part of the interface being implemented by this class.